common/core/web/tools/sentry-manager/src/index.ts

```diff
@@ -0,0 +1,137 @@
+    keymanPlatform: string;
+
+    static STANDARD_ALIASABLE_FILES = {
+      'keymanweb.js':    'keymanweb.js',
+      'kmwuibutton.js':  'kmwuibutton.js',
+      'kmwuifloat.js':   'kmwuifloat.js',
+      'kmwuitoggle.js':  'kmwuitoggle.js',
+      'kmwuitoolbar.js': 'kmwuitoolbar.js'
+      // Also add entries for the naming system used by Android and iOS - and map them to the EMBEDDED upload, not the std 'native' one.
+    }
+
+    static DEFAULT_OPTIONS = {
+      hostPlatform: "native-web"
+    }
+
+    constructor(options: Options = KeymanSentryManager.DEFAULT_OPTIONS) {
+      this.keymanPlatform = options.hostPlatform;
+    }
+
+    // If we've recognized one of our source files, 
+    aliasFilename(filename: string): string|null {
+      if(!this.mayAlias(filename)) {
+        return null;
+      }
+
+      filename = KeymanSentryManager.STANDARD_ALIASABLE_FILES[filename];
+
+      switch(location.protocol) {
+        case 'http:':
+          return 'http://' + location.host + '/' + filename;
+        case 'file:':
+          return 'file:///' + filename;
+        default:
+          return null;
+      }
+    }
+
+    mayAlias(filename: string): boolean {
+      return !!KeymanSentryManager.STANDARD_ALIASABLE_FILES[filename];
+    }
+
+    // Filters all expected but unnecessary path prefixes, affixes, and suffixes reported by Sentry from our products.
+    // This allows us to mask all different sorts of installations with a single uploaded path.
+    pathFilter(event: any) {
+      // Get the underlying JS error.
+      let exception = event.exception;
+
+      // Iterate through all wrapped exceptions.
+      for(let e of exception.values) {
+        for(let frame of e.stacktrace.frames) {
+          let URL = frame.filename as string;
+          let filename: string = '';
+          try {
+            filename = URL.substr(URL.lastIndexOf('/')+1);
+            if(this.mayAlias(filename)) {
+              // Make the file look like it's at the host's root so that the Sentry server will recognize it
+              // regardless of actual filepath position.
+              frame.filename = this.aliasFilename(filename) || frame.filename;
+            }
+          } catch {
+            // not sure what to do yet, but... don't make more errors in the error handler.  That's a bad idea.
+          }
+        }
+      }
+
+      return event;
+    }
+
+    attachEventMetadata(event: any) {
+      // Ensure that the 'extra' object exists.  (May not exist for synthetic/custom Errors.)
+      event.extra = event.extra || {};
+      event.extra.keymanState = window['keyman']['getDebugInfo']();
+      event.extra.keymanHostPlatform = this.keymanPlatform;
+      return event;
+    }
+
+    /**
+     * Pre-processes a Sentry event object to provide more metadata and enhance the Sentry server's
+     * ability to match the error against release artifacts.
+     * @param event A Sentry-generated event
+     */
+    eventPreparer(event: any) {
```

Comment:
This modifies the input variable. A little squirmy on this -- would be better to copy it and mutate the copy.

```suggestion
    prepareEvent(event: any) {
```

Reply:
😕 ... uh, if you say so.  Pass by reference is a pretty common convention though, isn't it?

[Deep copying isn't a simple matter in JS](https://stackoverflow.com/a/122704), and we don't own the type being copied.  The objects involved aren't exactly shallow, either.

That said, we do have a small custom deep-copy implementation as part of [keyboard-processor's code](https://github.com/keymanapp/keyman/blob/master/common/core/web/keyboard-processor/src/utils/deepCopy.ts) at present.  I would rather not add `keyboard-processor` as a dev-dependency of this package b/c of node_module link pathing issues, and I'd rather avoid WETness in the codebase as well.

That said, I've been wanting to split that function (along with the other util methods there and kmwstring.ts, at minimum) off into their own mini-package; they're the cause of most of the Web refactor's `<reference path=""/>` spaghetti and may help address some of those link pathing issues.  (kmwstring.ts is why the lm-layer links with keyboard-processor.)  All of that work, though, would be work far better left to a different PR - and one I hope to write soon.

Reply:
OK. In this case, I'm fine with that if we make it clear that we are mutating the parameter in the comments -- and make this return a boolean success value rather than the input parameter. Then future maintainers won't get confused.